whatwg-url PR: https://github.com/jsdom/whatwg-url/pull/71

All the new tests appear to pass, but a bunch of old ones have become invalidated, for the host and hostname setters. I'll only list host since hostname is analogous. As usual, "expected" = new WPTs, "actual" = jsdom/whatwg-url with the spec patch applied.

```
  1) Web platform tests setters host <sc://x/>.host = " " :

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///

  2) Web platform tests setters host <sc://x/>.host = "
" :

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///
" :) Web platform tests setters host <sc://x/>.host = "

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///

  4) Web platform tests setters host <sc://x/>.host = "#" :

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///

  5) Web platform tests setters host <sc://x/>.host = "/" :

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///

  6) Web platform tests setters host <sc://x/>.host = "?" :

      AssertionError: 'sc://x/' == 'sc:///'
      + expected - actual

      -sc://x/
      +sc:///

  7) Web platform tests setters host <view-source+http://example.net/foo>.host = "" The empty host is OK for non-special schemes:

      AssertionError: 'view-source+http://example.net/foo' == 'view-source+http:///foo'
      + expected - actual

      -view-source+http://example.net/foo
      +view-source+http:///foo
```

--- a/url/setters_tests.json
+++ b/url/setters_tests.json
@@ -264,6 +264,22 @@
                 "href": "http://%c3%89t%C3%A9@example.net/",
                 "username": "%c3%89t%C3%A9"
             }
+        },
+        {
+            "href": "sc:///",
+            "new_value": "x",
+            "expected": {
+                "href": "sc:///",
+                "username": ""
+            }
+        },
+        {
+            "href": "file://test/",
+            "new_value": "test",
+            "expected": {
+                "href": "file://test/",
+                "username": ""
+            }
         }
     ],
     "password": [
@@ -342,6 +358,22 @@
             "expected": {
                 "href": "http://:%c3%89t%C3%A9@example.net/",
                 "password": "%c3%89t%C3%A9"
+            }
+        },
+        {
+            "href": "sc:///",
+            "new_value": "x",
+            "expected": {
+                "href": "sc:///",
+                "password": ""
+            }
+        },
+        {
+            "href": "file://test/",
+            "new_value": "test",
+            "expected": {
+                "href": "file://test/",
+                "password": ""
             }
         }
     ],
@@ -759,6 +791,46 @@
                 "host": "example.net",
                 "hostname": "example.net"
             }
+        },
+        {
+            "href": "file://y/",
+            "new_value": "x:123",
+            "expected": {
+                "href": "file://y/",
+                "host": "y",
+                "hostname": "y",
+                "port": ""
+            }
+        },
+        {
+            "href": "file://y/",
+            "new_value": "loc%41lhost",
+            "expected": {
+                "href": "file:///",
+                "host": "",
+                "hostname": "",
+                "port": ""
+            }
+        },
+        {
+            "href": "sc://test@test/",
+            "new_value": "",
+            "expected": {
+                "href": "sc://test@test/",
+                "host": "test",
+                "hostname": "test",
+                "username": "test"
+            }
+        },
+        {
+            "href": "sc://test:12/",
+            "new_value": "",
+            "expected": {
+                "href": "sc://test:12/",
+                "host": "test:12",
+                "hostname": "test",
+                "port": "12"
+            }
         }
     ],
     "hostname": [
@@ -1043,6 +1115,46 @@
                 "host": "example.net",
                 "hostname": "example.net"
             }
+        },
+        {
+            "href": "file://y/",
+            "new_value": "x:123",
+            "expected": {
+                "href": "file://y/",
+                "host": "y",
+                "hostname": "y",
+                "port": ""
+            }
+        },
+        {
+            "href": "file://y/",
+            "new_value": "loc%41lhost",
+            "expected": {
+                "href": "file:///",
+                "host": "",
+                "hostname": "",
+                "port": ""
+            }
+        },
+        {
+            "href": "sc://test@test/",
+            "new_value": "",
+            "expected": {
+                "href": "sc://test@test/",
+                "host": "test",
+                "hostname": "test",
+                "username": "test"
+            }
+        },
+        {
+            "href": "sc://test:12/",
+            "new_value": "",
+            "expected": {
+                "href": "sc://test:12/",
+                "host": "test:12",
+                "hostname": "test",
+                "port": "12"
+            }
         }
     ],
     "port": [
@@ -1197,6 +1309,46 @@
                 "host": "example.net:8080",
                 "hostname": "example.net",
                 "port": "8080"
+            }
+        },
+        {
+            "href": "file://test/",
+            "new_value": "12",
+            "expected": {
+                "href": "file://test/",
+                "port": ""
+            }
+        },
+        {
+            "href": "file://localhost/",
+            "new_value": "12",
+            "expected": {
+                "href": "file:///",
+                "port": ""
+            }
+        },
+        {
+            "href": "non-base:value",
+            "new_value": "12",
+            "expected": {
+                "href": "non-base:value",
+                "port": ""
+            }
+        },
+        {
+            "href": "sc:///",
+            "new_value": "12",
+            "expected": {
+                "href": "sc:///",
+                "port": ""
+            }
+        },
+        {
+            "href": "sc://x/",
+            "new_value": "12",
+            "expected": {
+                "href": "sc://x:12/",
+                "port": "12"
             }
         }
     ],

--- a/url/urltestdata.json
+++ b/url/urltestdata.json
@@ -1020,6 +1020,26 @@
     "hash": ""
   },
   {
+    "input": "file://example:1/",
+    "base": "about:blank",
+    "failure": true
+  },
+  {
+    "input": "file://example:test/",
+    "base": "about:blank",
+    "failure": true
+  },
+  {
+    "input": "file://example%/",
+    "base": "about:blank",
+    "failure": true
+  },
+  {
+    "input": "file://[example]/",
+    "base": "about:blank",
+    "failure": true
+  },
+  {
     "input": "ftps:/example.com/",
     "base": "http://example.org/foo/bar",
     "href": "ftps:/example.com/",

